front/components/assistant/conversation/actions/AgentMessageActions.tsx

```diff
@@ -11,6 +11,24 @@ interface AgentMessageActionsProps {
   size?: MessageSizeType;
 }
 
+function isActionComplete(action: AgentActionType): boolean {
+  switch (action.type) {
+    case "retrieval_action":
+      return action.documents !== null;
+    case "tables_query_action":
+    case "process_action":
+    case "websearch_action":
+    case "browse_action":
+      return "output" in action ? action.output !== null : false;
+    case "dust_app_run_action":
+      return action.output !== null && action.runningBlock === null;
+    case "visualization_action":
+      return action.generation !== null;
+    default:
+      return false;
```

Comment:
Could we `assertNever(action);` so that if we had a new action type typescript will let us know there's this spot to update?